src/transformers/models/t5/modeling_t5.py

```diff
@@ -1309,6 +1314,13 @@ def forward(
                 raise ValueError("You have to initialize the model with valid token embeddings")
             inputs_embeds = self.embed_tokens(input_ids)
 
+            if self.project_scalars is not None:
+                for sample_idx, (_curr_scalar_ind, _curr_scalar_vals) in enumerate(zip(encoder_input_scalars_indices, encoder_input_scalars_values)):
+                    assert not ((_curr_scalar_ind is not None) ^ (_curr_scalar_vals is not None)) #must be neither or both
+                    if (_curr_scalar_ind is not None) and (_curr_scalar_ind.shape[0] > 0):
+                        inputs_embeds[sample_idx][_curr_scalar_ind] += self.project_scalars(_curr_scalar_vals[..., None])
```

Comment:
we can make it vectorized here, but it will be a bit "cheating" because it would mean that as preprocessing it was prepared as a single tensor, and there we had a loop on samples.

there is one possible strategy that we can use, if our collate() was able to also return a mask of valid locations.

either way, not doing this in this PR - let's include this in the PR that moves to calculating embedding outside the transformers and uses inputs_embeds in transformers